vtexdocs/devportal
# Feature/feedback section modal

#### What is the purpose of this pull request?

Add the modal that will appear when the user selects a feedback button.

[Documented Here :book: ](https://www.notion.so/vtexhandbook/Adicionar-modal-do-feedback-section-em-paginas-de-API-Guides-99ccf10b85d64c48a0745586878fcc98)

#### What problem is this solving?

The feedback section stays static. When the user selects a feedback button, a modal should appear with a `textarea` where the user can give us their feedback explanation. This PR implements the modal structure and logic. But, the logic for collecting and storing feedback will be implemented in future PRs.

#### How should this be manually tested?

Go to [this page](https://deploy-preview-49--elated-hoover-5c29bf.netlify.app/docs/api-guides/clients) and try:
* Select a feedback button (**yes** or **no**) and check:
   * The modal appears in the correct location: If there is space between the selected button and the top of the page, the modal should appear below it, otherwise, at the bottom. And check the arrow direction and position.
   * The page background is darkened
   * The scrollbar is locked, the user cannot scroll the page while the modal is appearing
   * The selected button stays `active` while the modal is appearing
* Click **outside** of the modal and check:
    * The background returned to normal
    * The scrollbar is enabled and the page scroll stays at the same place
    * The selected feedback button has returned to the default state
* Click on the `SEND FEEDBACK` button inside the modal and check:
    * The background returned to normal
    * The scrollbar is enabled and the page scroll stays at the same place
    * The selected button and the feedback section text have changed to the "submitted" appearance 

#### Screenshots or example usage

https://user-images.githubusercontent.com/42784961/173897926-109dccfb-ee86-4da1-9c8c-9547e1ed3246.mp4

#### Types of changes
